testsuite/cluster-test/src/prometheus.rs

```diff
@@ -100,9 +102,42 @@ impl Prometheus {
             .avg()
             .ok_or_else(|| format_err!("Failed to compute avg"))
     }
+
+    pub fn query_range_max(
+        &self,
+        query: String,
+        start: &Duration,
+        end: &Duration,
+        step: u64,
+    ) -> Result<HashMap<String, f64>> {
+        let response = self.query_range(query, start, end, step)?;
+        response
+            .max()
+            .ok_or_else(|| format_err!("Failed to compute max"))
+    }
 }
 
 impl MatrixResponse {
+    pub fn max(&self) -> Option<HashMap<String, f64>> {
```

Comment:
To be honest I am wondering why do you want to return `Option<HashMap<String, f64>> ` instead of just returning `HashMap<String, f64>>`?
If it's empty then return empty hash map